### DISCUSSION TITLE
configuring f-prime has no effect

### DISCUSSION THREAD

**Discussion by shalevm** [nasa/fprime#1620] [Q&A] [answered]
Hi all,
I've been trying to configure f-prime to better suit my project's needs, as per the instructions described [here](https://nasa.github.io/fprime/UsersGuide/dev/configuring-fprime.html).

specifically:
1. I created a **settings.ini** file with the following contents:
```
[fprime]
; path to the root of the fprime project
project_root: ..

; Path to the F´ framework root
framework_path: ../fprime

; Path to autocode constants ini file.
ac_constants: ./Cfg/AcConstants.ini

; Path to configuration header directory.
config_directory: Cfg
```
2. I created a **Cfg** folder.
3. I copied all the ***Cfg.hpp** files and the **AcConstants.ini** file into the **Cfg** folder.
4. in the file **CmdDispatcherImplCfg.hpp** i changed the value of the enum constant 
    `CMD_DISPATCHER_DISPATCH_TABLE_SIZE` 
    from 100 to 300.
5. I purged, regenerated, and rebuilt the project.

When i run my program this has no apparent effect: i still fail an assertion in line 54 of **CommandDispatcherImpl.cpp**
because I'm trying to register more then 100 commands.

Regardless, I also tried changing the constant 
`RateGroupDriverRateGroupPorts`
in the **AcConstants.ini** file from 3 to 5, again with no apparent affect.

Any ideas as to what I'm missing?
Thanks.

**Comment by ThibFrgsGmz** [comment]
Like you, I tried to temporarily register over 150 commands and found this issue. I reported it in this ticket: https://github.com/fprime-community/fprime-tools/issues/96

**Reply by shalevm** [reply]
Thanks, good to know it's not just me :)

**Reply by shalevm** [reply]
@LeStarch any news on this issue? the above-mentioned ticket by @ThibFrgsGmz has gotten no response... 
Thanks.

**Reply by LeStarch** [reply]
I'll take a look and report back.

**Comment by ebenezerarunkumar** [comment]
Are you using the most recent release of fprime-tools? https://github.com/fprime-community/fprime-tools/releases/tag/v3.1.1

This issue was fixed for me when I upgraded fprime-tools. Hopefully it should fix it for you as well!

**Reply by shalevm** [reply]
Thanks @EbenezerA99 
I'm fairly new to f-prime - I'm just using the main f-prime repo... I wasn't familiar with the fprime-tools repo - is it necessary for development? 
if so - do I just clone it and reference it in the **settings.ini** file with `library_locations` ? or do I just "pip install" it?

**Reply by shalevm** [reply]
In any case, my **settings.ini** has a `config_directory` rather than a `config_dir`, which, if i understand correctly, implies that the bug that was fixed does not apply to me.

**Reply by ebenezerarunkumar** [reply]
hmmm, when you updated the constants, did you update them in the `AcConstants.ini `file or the `AcConstants.fpp` file? for my deployments, it seems like the changes only took effect if I changed the `.fpp` file and not the `.ini` file

**Reply by shalevm** [reply]
well, the f-prime manual doesn't specify that you need a copy of `AcConstants.fpp`, only `AcConstants.ini`. in any case, my current problem is with the component constants defined in the `*Cfg.hpp` files, not the framework constants.
oh, and by the way, I'm not sure installing a new version of fprime-tools independently of the main f-prime repo is well advised - fprime expects a specific version of fprime-tools, and will not build or even generate with a different version. so I guess the way to go is to update the fprime repo (assuming it's been updated to use the new fprime-tools version)?

**Reply by ebenezerarunkumar** [reply]
hmmm while fprime expects a  specific version of the tools, my deployments still build and generate with the newer version of fprime-tools and fprime-gds

**Reply by shalevm** [reply]
how strange... mine refused to do so... i had to reinstall version 3.1.0
hopefully I'll get an official answer from the dev team soon enough.
thanks for you help anyways :)

**Comment by ebenezerarunkumar** [comment]
No worries! Fprime-tools is needed to develop with fprime, but it gets automatically installed whenever you install fprime(it is listed in the requirements.txt file).

To update the fprime-tools version, no need to clone the repo, you can just run `pip install fprime-tools==3.1.1`

**Comment by LeStarch** [comment, marked as answer]
I looked into this and the error is indeed solved by the toolset `v3.1.1`.  I will bump that pointer in master before the point release we are preparing.  Here is what I did to test:

In my custom config:  (Project/config)
```
CMD_DISPATCHER_DISPATCH_TABLE_SIZE = -1,
```

In `settings.ini`:
```
config_directory: ./config
```

Building with `requirements.txt` did not error with the -1 size, nor did the `generate` step produce correct paths.  Then I followed the upgraded installation `pip install fprime-tools==v3.1.1` and repeated `generate` and `build` seeing the updated paths and the -1 size invalid error from the compiler.

We'll bump the version of tools in our `requirements.txt` file before the point-release.  If you have trouble with the upgrade, let me know so we can help!

**Reply by shalevm** [reply]
Thanks Michael!!

**Reply by LeStarch** [reply]
https://github.com/nasa/fprime/pull/1622  should help.
